pubsub/pulsar/metadata.go

```diff
@@ -26,12 +26,20 @@ type pulsarMetadata struct {
 	Tenant                  string                    `mapstructure:"tenant"`
 	Namespace               string                    `mapstructure:"namespace"`
 	Persistent              bool                      `mapstructure:"persistent"`
-	Token                   string                    `mapstructure:"token"`
 	RedeliveryDelay         time.Duration             `mapstructure:"redeliveryDelay"`
 	internalTopicSchemas    map[string]schemaMetadata `mapstructure:"-"`
 	PublicKey               string                    `mapstructure:"publicKey"`
 	PrivateKey              string                    `mapstructure:"privateKey"`
 	Keys                    string                    `mapstructure:"keys"`
+
+	AuthType         string   `mapstructure:"authType"`
```

Comment:
Although I'm open to adding the (optional) "authType" metadata back, I do not support the other change of refreshing tokens in background when not needed. Since clients are long-lived and cached, the perf benefit is negligible. At the same time, refreshing tokens in background when we are not going to use them is wasteful (requires another goroutine, etc) and not polite.